expr/func_aggregate.go

```diff
@@ -9,21 +10,35 @@ import (
 )
 
 type FuncAggregate struct {
-	in  []GraphiteFunc
-	agg seriesAggregator
+	in           []GraphiteFunc
+	name         string
+	xFilesFactor float64
 }
 
 // NewAggregateConstructor takes an agg string and returns a constructor function
-func NewAggregateConstructor(aggDescription string, aggFunc crossSeriesAggFunc) func() GraphiteFunc {
+func NewAggregateConstructor(name string) func() GraphiteFunc {
 	return func() GraphiteFunc {
-		return &FuncAggregate{agg: seriesAggregator{function: aggFunc, name: aggDescription}}
+		return &FuncAggregate{name: name}
 	}
 }
 
+func NewAggregate() GraphiteFunc {
+	return &FuncAggregate{}
+}
+
 func (s *FuncAggregate) Signature() ([]Arg, []Arg) {
-	return []Arg{
-		ArgSeriesLists{val: &s.in},
-	}, []Arg{ArgSeries{}}
+	if s.name == "" {
+		return []Arg{
+			ArgSeriesLists{val: &s.in},
+			ArgString{val: &s.name, validator: []Validator{IsAggFunc}, key: "func"},
+			ArgFloat{val: &s.xFilesFactor, opt: true, key: "xFilesFactor"},
```

Comment:
Doesn't this input need to be validated? 
https://github.com/bloomberg/metrictank/pull/83/files#diff-6688f16342c5fcf0469dc1a21fd67adbR23